File: Frontend/src/components/services/books.js
import axios from "axios";

class Books {
    static async getBooks(
        page = 1,
        limit = 10,
        categoryId = null,
        ageGroupId = null
    ) {
        const res = await axios.get("http://localhost:8000/api/book", {
            params: {
                page,
                limit,
                category_id: categoryId,
                age_group_id: ageGroupId   
            },
            withCredentials: true,
            headers: { apiKey: "123456789apikeysecure" }
        });

        return res.data;
    }

    static async getRandomBooks(limit = 10) {
        const res = await axios.get(
            "http://localhost:8000/api/book/random/limit",
            {
                params: { limit },
                withCredentials: true,
                headers: { apiKey: "123456789apikeysecure" }
            }
        );
        return res.data;
    }

    static async getBookById(id) {
        const res = await axios.get(
            `http://localhost:8000/api/book/${id}`,
            {
                withCredentials: true,
                headers: { apiKey: "123456789apikeysecure" }
            }
        );
        return res.data;
    }
}

export default Books;

File: Backend/routes/books.py
from fastapi import APIRouter, Query,Depends
from utils.auth_helper import get_user
from services.bookService import (
    get_random_books,
    get_book_by_id,
    get_books
)

router = APIRouter()

@router.get("/")
def get_books_route(
    page: int = Query(1, ge=1),
    limit: int = Query(8, ge=1, le=50),
    category_id: int | None = Query(None),
    age_group_id: int | None = Query(None),
    user=Depends(get_user)
):
    return get_books(page, limit, category_id, age_group_id)



@router.get("/random/limit")
def get_random_books_route(limit: int = 10, user=Depends(get_user)):
    return get_random_books(limit)


@router.get("/{book_id}")
def get_book_by_id_route(book_id: int, user=Depends(get_user)):
    return get_book_by_id(book_id)
